app/assets/javascripts/controllers/ops/pglogical_replication_form_controller.js

```diff
@@ -0,0 +1,284 @@
+    $http.get('/ops/pglogical_subscriptions_form_fields/' + pglogicalReplicationFormId).success(function(data) {
+      $scope.pglogicalReplicationModel.replication_type = data.replication_type;
+      $scope.pglogicalReplicationModel.subscriptions = angular.copy(data.subscriptions);
+
+      if ($scope.pglogicalReplicationModel.replication_type == "none")
+        miqService.miqFlash("warn", __("No replication role has been set"));
+
+      $scope.afterGet = true;
+      $scope.modelCopy = angular.copy( $scope.pglogicalReplicationModel );
+      miqService.sparkleOff();
+    });
+
+    $scope.$watch("pglogicalReplicationModel.subscriptions", function() {
+      $scope.form = $scope.angularForm;
+    });
+  };
+
+  var pglogicalManageSubscriptionsButtonClicked = function(buttonName, serializeFields) {
+    miqService.sparkleOn();
+    var url = '/ops/pglogical_save_subscriptions/' + pglogicalReplicationFormId + '?button=' + buttonName;
+    miqService.miqAjaxButton(url, serializeFields);
+  };
+
+  $scope.resetClicked = function() {
+    $scope.pglogicalReplicationModel = angular.copy( $scope.modelCopy );
+    $scope.angularForm.$setUntouched(true);
+    $scope.angularForm.$setPristine(true);
+    miqService.miqFlash("warn", __("All changes have been reset"));
+  };
+
+  $scope.saveClicked = function() {
+    // remove existing subscriptions that have not changed before sending them up for save
+    $scope.pglogicalReplicationModel.subscriptions.forEach(function(subscription, index, object) {
+      if (typeof subscription.id !== 'undefined' && !subscriptionChanged(subscription, $scope.modelCopy.subscriptions[index])) {
+        object.splice(index, 1);
+      }
+    });
+    pglogicalManageSubscriptionsButtonClicked('save', {
+      'replication_type': $scope.pglogicalReplicationModel.replication_type,
+      'subscriptions' : $scope.pglogicalReplicationModel.subscriptions
+    });
+    $scope.angularForm.$setPristine(true);
+  };
+
+  // check if subscription values have been changed
+  var subscriptionChanged = function(new_subscription, original_subscription) {
+    if (new_subscription.dbname   === original_subscription.dbname &&
+        new_subscription.host     === original_subscription.host &&
+        new_subscription.user     === original_subscription.user &&
+        new_subscription.password === original_subscription.password &&
+        new_subscription.port     === original_subscription.port)
+      return false;
+    else
+      return true;
+  }
+
+  // replication type changed, show appropriate flash message
+  $scope.replicationTypeChanged = function() {
+    miqService.miqFlashClear();
+    var original_value = $scope.modelCopy.replication_type;
+    var new_value      = $scope.pglogicalReplicationModel.replication_type;
+    if ((original_value == "none" && new_value == "none") || (original_value == "remote" && new_value == 'none'))
+      miqService.miqFlash("warn", __("No replication role has been set"));
```

Comment:
This message seems out of place compared to the others.
The others are all future tense (i.e "x *will* happen when you hit save") this one seems to be present tense.
Maybe something like "Replication will be disabled for this region"?